custom_components/versatile_thermostat/thermostat_climate.py

```diff
@@ -719,6 +719,22 @@ async def end_climate_changed(changes: bool):
             )
             return
 
+        # Forget event when the new target temperature is out of range
+        if (
+            not new_target_temp is None
+            and not self._attr_min_temp is None
+            and not self._attr_max_temp is None
+            and (new_target_temp < self._attr_min_temp or new_target_temp > self._attr_max_temp)
```

Comment:
This can be safely simplified with:

and self._attr_max_temp < new_target_temp < self._attr_min_temp

This is a specific Python syntax !

Reply:
Thanks for the hint, now this line is pretty!